[Quant][docs] Add README for BackendConfig

andrewor14 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86523

Summary: This adds a README for `torch.ao.quantization.backend_config`
that describes both the high level motivation and the specifications
of the BackendConfig API.

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

andrewor14 committed:
[Quant[docs]] Add README for BackendConfig

Summary: This adds a README for `torch.ao.quantization.backend_config`
that describes both the high level motivation and the specifications
of the BackendConfig API.

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

[ghstack-poisoned]

andrewor14 committed:
Update on "[Quant][docs] Add README for BackendConfig"


Summary: This adds a README for `torch.ao.quantization.backend_config`
that describes both the high level motivation and the specifications
of the BackendConfig API.

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

[ghstack-poisoned]

andrewor14 committed:
Update on "[Quant][docs] Add README for BackendConfig"


Summary: This adds a README for `torch.ao.quantization.backend_config`
that describes both the high level motivation and the specifications
of the BackendConfig API.

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

[ghstack-poisoned]

jerryzh168 commented on torch/ao/quantization/backend_config/README.md:
````diff
@@ -17,4 +41,110 @@ tensor_1            tensor_2
+    .set_backend_pattern_config(linear_config) \
+    .set_backend_pattern_config(conv_relu_config)
+```
+#### set_observation_type
````
I feel some of these should be moved to docstrings for the API?

jerryzh168 replied:
I think we should discuss about the goal for the README, my current understanding is that the README should describe the design decisions and how it is implemented

vkuzo replied:
I think a key thing would be to avoid duplication.  Either we should have these examples in docstrings or here, but not in both places.  If it's duplicated the risk of it being outdated increases.

vkuzo replied:
I do feel like what is in this file now would fit more naturally in documentation, this seems to be documentation for users of BackendConfig, not necessarily developers of AO code.

andrewor14 committed:
Update on "[Quant][docs] Add README for BackendConfig"


Summary: This adds a README for `torch.ao.quantization.backend_config`
that describes both the high level motivation and the specifications
of the BackendConfig API.

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

[ghstack-poisoned]

jerryzh168 reviewed (approved):
LGTM, thanks!

andrewor14 committed:
Update on "[Quant][docs] Add README for BackendConfig"


Summary: This adds a README for `torch.ao.quantization.backend_config`
that describes both the high level motivation and the specifications
of the BackendConfig API.

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

[ghstack-poisoned]

andrewor14 committed:
Update on "[Quant][docs] Add README for BackendConfig"


Summary: This adds a README for `torch.ao.quantization.backend_config`
that describes both the high level motivation and the specifications
of the BackendConfig API.

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

[ghstack-poisoned]

andrewor14 committed:
Update on "[Quant][docs] Add README for BackendConfig"


Summary: This adds a README for `torch.ao.quantization.backend_config`
that describes both the high level motivation and the specifications
of the BackendConfig API.

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

[ghstack-poisoned]

andrewor14 committed:
Update on "[Quant][docs] Add README for BackendConfig"


Summary: This adds a README for `torch.ao.quantization.backend_config`
that describes both the high level motivation and the specifications
of the BackendConfig API.

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

[ghstack-poisoned]

andrewor14 committed:
Update on "[Quant][docs] Add README for BackendConfig"


Summary: This adds a README for `torch.ao.quantization.backend_config`
that describes both the high level motivation and the specifications
of the BackendConfig API.

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

[ghstack-poisoned]

andrewor14 committed:
Update on "[Quant][docs] Add README for BackendConfig"


Summary: This adds a README for `torch.ao.quantization.backend_config`
that describes both the high level motivation and the specifications
of the BackendConfig API.

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

[ghstack-poisoned]
